openhcl/hcl/src/ioctl.rs

```diff
@@ -1961,33 +1964,6 @@ impl Hcl {
         // Open the hypercall pseudo-device
         let mshv_hvcall = MshvHvcall::new()?;
 
-        // TODO SNP: When it's checked in, the isolation type should instead be queried
-        // from the kernel.
-        let isolation = if cfg!(guest_arch = "x86_64") {
-            // xtask-fmt allow-target-arch cpu-intrinsic
-            #[cfg(target_arch = "x86_64")]
-            {
-                let result = safe_x86_intrinsics::cpuid(
```

Comment:
rather than deleting this check entirely, maybe we should keep it around as a sanity check (albeit outside the core openhcl init path, for reasons you discussed in the PR description)

Reply:
Good idea